### ISSUE TITLE
F6 and shift F6 to navigate across workbench

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#94921]
These command would be called "Focus Next Part" and "Focus Previous Part" and the focus would move cyclicaly in the following proposed order:
Editor -> Status Bar > Side Bar > Panel > Editor

Distilled from https://github.com/microsoft/vscode/issues/94677

**Comment by isidorn** [comment]
fyi @bpasero for awarness and maybe better command name ideas.
Also I plan to add `setFocus(part: Parts)` to the `IWorkbenchLayoutService`. Let me know if you would like me to create a PR.

**Comment by bpasero** [comment]
Shouldn't the layout service take care of what is next or previous and then the method would need an argument to focus next or previous? 

Maybe coordinate with @sbatten as layout owner.

**Comment by isidorn** [comment]
@bpasero that also works for me.
@sbatten I will create a PR and you can give it a look if you would like.

**Comment by isidorn** [comment]
Just found out that we already have some similar navigation actions, so I will make the changes there
https://github.com/microsoft/vscode/blob/1b9a442590e17687da1705cdebc101763cba2b6f/src/vs/workbench/browser/actions/navigationActions.ts#L209

**Comment by isidorn** [comment]
After discussing with @sbatten we decided to go with the logic living in the action, so we make the things in the `layoutService` more simple and only add a `setFocus(part)`. I have pushed that commit, @sbatten and @bpasero feedback welcome.

@bpasero Please note that I have changed the status bar part to give it `tabIndex = -1` which means not focussable via tab but focussable via explicit javascript calls which is exactly what we need.

@marcozehe @leonardder @marlon-sousa try it out with vscode insiders from Thursday and let me know how this behaves for you. Please note that once status bar is focused there is no navigation within the status bar, in case that is needed please let me know and we can tackle via another issue.
Also if the focus is some unrelated part F6 and shift + F6 will always focus the editor (that made sense to me, but we can change it of course)

**Comment by leonardder** [comment]
FYI @bramd

**Comment by jvesouza** [comment]
@isidorn  I tested it and in my opinion it was very good, congratulations! 
I may be wrong, but it seems to me that there should be a way to interact with the status bar using the arrows, tab and enter since the focus is on it. Currently I can interact using the orca flat review mode, but for me it would be much more natural to click on an item in the status bar using the enter key or the space bar. Let's see what other users have to say.

@joanmarie When I press the f6 key the orca announces something like:
'complementary content' or
'leaving complementary content.'
I believe that these messages could be omitted, if possible.

**Comment by isidorn** [comment]
@jvesouza yes arrow navigation in the status bar makes sense to me. Let's see if more users want this and if yes we can file a feature request for this and we could look into it.

Since you gave positive feedback I am adding verified label here.
Note that if a part is hidden we simply skip navigating over it. I think that makes sense.

**Comment by MarcoZehe** [comment]
I agree, having the status bar items interactive would be cool. There is precedent for this in MS Word, for example.

As for the leaving and entering landmark announcements: NVDA omits the "leaving" part for landmarks, so the announcement is actually quite fluent. Perhaps Orca could take a bite or two out of how NVDA deals with ancestor traversal announcements. And for that matter, VoiceOver, too, because that is also very chatty in VS Code. And it's not VS Code's fault.

**Comment by isidorn** [comment]
Makes sense. Created https://github.com/microsoft/vscode/issues/95433 to track keyboard navigation in the status bar.

**Comment by leonardder** [comment]
I noticed that I had a terminal active in VS Code insiders. I wasn't able to get out of that with f6 / shift+f6. I think I'd expect it to.

**Comment by jvesouza** [comment]
@leonardder I tested it here using orca and it worked without problems. I activated a terminal by pressing ctrl + ` and then used the f6 key to return to the editor.

**Comment by isidorn** [comment]
@leonardder I can repro that on macOS and I know you are on Windows.
So let's check with @Tyriar why does the temrinal take away the `F6` and `shift + F6` key on windows and macOS?

**Comment by leonardder** [comment]
> 
> 
> @leonardder I tested it here using orca and it worked without problems. I activated a terminal by pressing ctrl + ` and then used the f6 key to return to the editor.

Interesting enough, if I follow that approach, it works for me as well. However as soon as I restart Code Insiders, f6 is eaten by the terminal and doesn't move focus any more.

**Comment by jvesouza** [comment]
@isidorn  I cannot reproduce exactly in linux but I found another one that may be related to the problem described by @leonardder .
I activate the terminal by pressing ctrl`, then press f6 and press ctrl + `again to return to the terminal.
When I do this, a character that I cannot identify appears at the prompt. If I press the enter key this character is sent to the shell and the shell returns an error message.
I can try to elaborate the test case better if necessary.

**Comment by joanmarie** [comment]
@MarcoZehe: Does NVDA ever announce leaving messages for landmarks?

@jvesouza: Have you tried disabling those announcements in Orca's app specific preferences for VSCode (on the speech preferences page)? Or do you want the entering messages spoken, but just the leaving messages eliminated?

**Comment by MarcoZehe** [comment]
@joanmarie No, not for landmarks. It does so for other things such as lists or tables, but not landmarks, and I don't think regions, either, but am not a 100% positive on the latter.

**Comment by jvesouza** [comment]
@joanmarie  After I disabled Announce landmarks during navigation the messages were no longer announced. To be honest, I hadn't thought about that option.
Thanks.

**Comment by joanmarie** [comment]
> @joanmarie After I disabled Announce landmarks during navigation the messages were no longer announced. To be honest, I hadn't thought about that option.
> Thanks.

My pleasure. Do you see that as the appropriate solution for this problem? Or should Orca only present those messages when in browse mode? Note that if you think the latter, it would apply to Firefox and Chrome in addition to VSCode.

**Comment by jvesouza** [comment]
@joanmarie For me this solution is more than perfect, since we can disable or enable the option by application.

**Comment by bramd** [comment]
Works great here. Looking forward to status bar keyboard navigation as well.

**Comment by Tyriar** [comment]
I just tried this out, fixed it by making the workbench.action.focusNextPart and workbench.action.focusPreviousPart commands skip the terminal by default.

**Comment by isidorn** [comment]
@Tyriar thanks a lot! Really appreciate it.
